Sources/SwiftProtobuf/Message+JSONAdditions.swift

```diff
@@ -63,17 +64,19 @@ extension Message {
   /// serialized message in JSON format.
   ///
   /// - Parameter jsonString: The JSON-formatted string to decode.
+  /// - Parameter extensions: An ExtensionMap for looking up extensions by name
   /// - Parameter options: The JSONDecodingOptions to use.
   /// - Throws: `JSONDecodingError` if decoding fails.
   public init(
     jsonString: String,
+    extensions: ExtensionMap = SimpleExtensionMap(),
```

Comment:
Hm, I know I suggested making these non optional, but looking back at the whole PR, I'm realizing in the binary apis we do use nil to indicate no map.  So now I'm torn.  Oh well, up to you for which makes the code cleaner, etc.  (one option might be to nil it here but default it to a SimpleExtensionMap in that case in the actually Decoder/Visitors so they don't have to check in all case; anyway, up to you)

Reply:
Hmmm...  Looks like both binary and textformat use an optional here and at the decoder initializer.  So I've converted the JSON to use the same pattern.  Down at the scanner where the extension map is actually stored, I now fill in an empty SimpleExtensionMap() to avoid nil checks.  (Since `SimpleExtensionMap()` is just a wrapper around a dictionary and empty dictionaries initialize trivially, this is no big deal.)